addrmgr/addrmanager.go

```diff
@@ -991,13 +1098,16 @@ func (a *AddrManager) SetServices(addr *wire.NetAddress, services wire.ServiceFl
 		ka.na = &naCopy
 		ka.mtx.Unlock()
 	}
+	return nil
 }
 
 // AddLocalAddress adds na to the list of known local addresses to advertise
 // with the given priority.
+//
+// This function is safe for concurrent access.
 func (a *AddrManager) AddLocalAddress(na *wire.NetAddress, priority AddressPriority) error {
 	if !IsRoutable(na) {
-		return fmt.Errorf("address %s is not routable", na.IP)
+		return fmt.Errorf("address %s is not routable", ipString(na))
```

Comment:
Perhaps it's coming up in a future commit, but I really think the new type that is local to the address manager should end up with a `fmt.Stringer` on it so the address can just be printed like normal.

Reply:
The new type satisfies the `fmt.Stringer` interface.

Reply:
if that's the case then `ipString` helper would not be needed to stringify `na`, just providing `na` should be adequate.

Reply:
Right, in the later commits that take advantage of the stringer interface the `ipString` call is dropped.  Two things to note here are that `ipString` converts TORv2 addresses to a friendly .onion hostname rather than printing as an IPv6 address, and that in the later commit `String` includes both the host and port rather than just the IP address.  My thinking here is that:

1. If the address is a tor address, it should print the ".onion" formatting rather than the IPv6 formatting (this commit)
2. The net address is uniquely identifiable by host:port so it should be printed with the port rather than just the host (handled by `String()` on the net address type in a later commit)

Plus, there is some overlap with the naming & behavior (of host:port) so the netaddress type can satisfy the net.Addr interface by adding a `Network` method when needed and switching that value based on the address type.